lib/curl_sha512_256.c

```diff
@@ -0,0 +1,596 @@
+/***************************************************************************
+ *                                  _   _ ____  _
+ *  Project                     ___| | | |  _ \| |
+ *                             / __| | | | |_) | |
+ *                            | (__| |_| |  _ <| |___
+ *                             \___|\___/|_| \_\_____|
+ *
+ * Copyright (C) Evgeny Grin (Karlson2k), <k2k@narod.ru>.
+ *
+ * This software is licensed as described in the file COPYING, which
+ * you should have received as part of this distribution. The terms
+ * are also available at https://curl.se/docs/copyright.html.
+ *
+ * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+ * copies of the Software, and permit persons to whom the Software is
+ * furnished to do so, under the terms of the COPYING file.
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+ * KIND, either express or implied.
+ *
+ * SPDX-License-Identifier: curl
+ *
+ ***************************************************************************/
+
+#include "curl_setup.h"
+
+#if !defined(CURL_DISABLE_DIGEST_AUTH) && !defined(CURL_DISABLE_SHA512_256)
+
+#include "curl_sha512_256.h"
+#include "warnless.h"
+
+/* ** This implementation of SHA-512/256 hash calculation was originally ** *
+ * ** written by Evgeny Grin (Karlson2k) for GNU libmicrohttpd.          ** *
+ * ** The author ported the code to libcurl. The ported code is provided ** *
+ * ** under curl license.                                                ** *
+ * ** This is a minimal version with minimal optimisations. Performance  ** *
+ * ** can be significantly improved. Big-endian store and load macros    ** *
+ * ** are obvious targets for optimisation.                              ** *
+ * ** The code is designed in a way so any wide enough type (64 bits or  ** *
+ * ** more) can be used for calculations. Potentially it can be larger   ** *
+ * ** then 64 bits.                                                      ** */
+
+#ifdef SIZEOF_INT
+#if SIZEOF_INT == 8
+#define CURLX_UINT64 unsigned int
```

Comment:
We use the `curlx_` prefix for some functions we provide source access to, for the curl tool. In this code you seem to use it for most things but for a different purpose. I would appreciate if you did not. For symbols that are private in this file, you don't really need any special prefix.

Reply:
OK.
These macros and functions could be moved to another file to be re-used.
I typically use some prefix to form a kind of namespace in C. This way I ensure that code will not conflict with any other code, both from the same application/library and from any third-party dynamic/static libs. Especially important for portability, when compiler and linker cannot hide non-exportable symbols (or file format does not allow symbols hiding).
Also this prevents any potential shadowing warning.
But since the prefix is reserved for other needs, I'll change it.